pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -171,6 +171,13 @@ func (r *PyTorchJobReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{Requeue: true, RequeueAfter: t}, nil
 	}
 
+	// if we create a pytorchjob, operator will create a podgroup later,if there is no enough resources in cluster, the podgroup will be pending, the reconcile function will break.
+	// if there are enough resources available later, the status of podgroup will be inqueue. then the reconcile function should continue to run. but the operator have not listening the podgroup
+	// if status of job is Created, it may mean that the operator have not created pods and services, we should reconcile job a little seconds later
+	if len(pytorchjob.Status.Conditions) == 1 && pytorchjob.Status.Conditions[0].Type == commonv1.JobCreated {
```

Comment:
Shouldn't we listen to pod group changes instead of this?

Reply:
@johnugeorge I have tried it, while some contributors think that it is not necessary for kubeflow operator to listen a thirdparty  crd(volcano podgroup). @zw0610

Reply:
> @johnugeorge I have tried it, while some contributors think that it is not necessary for kubeflow operator to listen a thirdparty crd(volcano podgroup). @zw0610

That's right. I would prefer training operator to remain neutral regarding gang-schedulers.

Reply:
In Katib, there is a similar feature where resources to be watched are taken as argument. https://github.com/kubeflow/katib/blob/master/cmd/katib-controller/v1beta1/main.go#L62   Resource to be watched are not hardcoded in the code-https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/trial/trial_controller.go#L102